### PR TITLE
fix(driver): fix some verifier issues in the modern probe

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area driver-modern-bpf

**Does this PR require a change in the driver versions?**

No

**What this PR does / why we need it**:

This PR presents 2 patches:

* (1° commit): `bpf_core_enum_value(enum bpf_func_id, BPF_FUNC_get_current_task_btf)` cannot be used if `BPF_FUNC_get_current_task_btf` is not defined in the kernel for this reason we first need to check is existence with `bpf_core_enum_value_exists(enum bpf_func_id, BPF_FUNC_get_current_task_btf)`

* (2° commit and 3° commit) Some kernels like `5.8.18` and `5.9.16` don't work with the actual ringbuf approach. The actual approach doesn't check if we are overcoming the ringbuf `reserved_size` this is because recent kernel verifiers `>=5.10` are able to keep track of the free space without the programmer's help. Unfortunately, this is not true on old kernels, for this reason, we need to help the verifier with this `CHECK_RINGBUF_SPACE` macro

> __Please note__:  I've changed the signature of `ringbuf__store_event_header` since passing the `event_size` is no more necessary, for this reason, I've changed a lot of files but in the end, the fix is straightforward :)  I suggest reviewing it commit by commit

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix(driver): fix some verifier issues in the modern probe
```

### CODE REVIEW

File: driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/fcntl.bpf.c [hunk 1]
```diff
@@ -20,6 +20,7 @@ int BPF_PROG(fcntl_e,
 		return 0;
 	}
 
+	/// TODO: we can remove the size from here.
```
Comment (FedeDP) [inline driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/fcntl.bpf.c]: So, can we drop all `SYSCALL_{E/X}_SIZE` defines?

Comment (Andreagit97) [reply driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/fcntl.bpf.c]: yeah this is exactly what I do in the next commit but only in `ringbuf__store_event_header` method, we need to keep the one in `ringbuf__reserve_space`

**Review by FedeDP — APPROVED** [review]
/approve

**Comment by hbrueckner** [comment]
@Andreagit97 very nice PR to make the verifier happy and make things easier :top: 

/lgtm

**Review by LucaGuerra — APPROVED** [review]
lgtm

**Pull request merged**